File: 프로그래머스/프린터.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        PriorityQueue<Integer> queue = new PriorityQueue<Integer>(Collections.reverseOrder());
        for(int n: priorities){
            queue.offer(n);
        }
        
        while(!queue.isEmpty()){
            for(int i = 0; i < priorities.length; i++){
                if(queue.peek() == priorities[i]){
                    answer++;
                    queue.poll();
                    if(i == location){
                        return answer;
                    }

                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/프린터.py
import heapq
def solution(priorities, location):
    answer = 0
    heap = []
    for num in priorities:
        heapq.heappush(heap, -num)
    while heap:
        for i in range(len(priorities)):    
            if -heap[0] == priorities[i]:
                heapq.heappop(heap)
                answer += 1
                if i == location:
                    return answer
    return answer